build.sbt

```diff
@@ -89,7 +89,47 @@ lazy val commonSettings = Seq(
   },
   // given that classDirectory is overriden to be _outside_ of target directory, we have
   // to make sure its being cleaned properly
-  cleanFiles += (classDirectory in Compile).value
+  cleanFiles += (classDirectory in Compile).value,
+
+  copyrightString := "Copyright 2002-2013, LAMP/EPFL",
+
+  resourceGenerators in Compile += generateVersionPropertiesFile.taskValue,
+
+  generateVersionPropertiesFile := {
+    val propFile = (resourceManaged in Compile).value / s"${name.value}.properties"
+    val props = new java.util.Properties
+
+    val (ver, mavenVer, osgiVer) =
+      version.value.split("-").toList match {
```

Comment:
Have you considered expressing this logic using regexp matchers? You could even inline regexps into patterns using string interpolation if that helps readability. Check out this article for details: https://github.com/OlegIlyenko/hacking-scala-blog/blob/master/posts/String-Interpolation-Meets-Regular-Expressions.md